File: back-end/src/main/java/com/twuc/shopping/bo/Product.java
package com.twuc.shopping.bo;

import com.twuc.shopping.po.ProductPO;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

import javax.validation.constraints.NotNull;

@Data
@Builder
@AllArgsConstructor
public class Product {
    @NotNull
    private String name;
    @NotNull
    private double price;
    @NotNull
    private String unit;
    @NotNull
    private String image;

    public Product(ProductPO productPO) {
        this.name = productPO.getName();
        this.price = productPO.getPrice();
        this.unit = productPO.getUnit();
        this.image = productPO.getImage();
    }
}

File: front-end/src/components/store/Product.js
import React from 'react';

const Product = props => (
  <div id={props.product.name}>
    <img src={props.product.image} alt={props.product.name} />
    <h4>{props.product.name}</h4>
    <p>单位:{props.product.price}元/{props.product.unit}</p>
    <button>+</button>
  </div>
);

export default Product;
